src/main.cpp

```diff
@@ -101,6 +114,422 @@ int pathfindertest(){
     return 0;
 }
 
+void get_vertex_data(std::vector<unsigned short> &indices,
+					 std::vector<glm::vec3> &indexed_vertices,
+					 std::vector<glm::vec3> &indexed_colors,
+					 std::vector<glm::vec3> &indexed_normals){
+	const char * path = "../SavedTerrain/pathfinder_input_1.qb";
+    World world(path);
+
+	world.Get_Mesh_Greedy(indices, indexed_vertices, indexed_colors, indexed_normals);
+	}
+
+int GUITest( void ) {
+	// Initialise GLFW
+	if( !glfwInit() )
+	{
+		fprintf( stderr, "Failed to initialize GLFW\n" );
+		getchar();
+		return -1;
+	}
+
+	glfwWindowHint(GLFW_SAMPLES, 4);
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+	glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE); // To make MacOS happy; should not be needed
+	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+
+	// Open a window and create its OpenGL context
+	window = glfwCreateWindow( 1024, 768, "Tutorial 16 - Shadows", NULL, NULL);
+	if( window == NULL ){
+		fprintf( stderr, "Failed to open GLFW window. If you have an Intel GPU, they are not 3.3 compatible. Try the 2.1 version of the tutorials.\n" );
+		getchar();
+		glfwTerminate();
+		return -1;
+	}
+	glfwMakeContextCurrent(window);
+    
+    // We would expect width and height to be 1024 and 768
+    int windowWidth = 1024;
+    int windowHeight = 768;
```

Comment:
Again, use defines, preferably named `INITIAL_WINDOW_*`

Reply:
I also did that, but these are nolonger user imputs